test/gtest/uct/ib/test_sockaddr.cc

```diff
@@ -855,6 +839,214 @@ UCS_TEST_P(test_uct_cm_sockaddr, connect_client_to_server_reject_with_delay)
 UCT_INSTANTIATE_SOCKADDR_TEST_CASE(test_uct_cm_sockaddr)
 
 
+class test_uct_cm_sockaddr_stress : public test_uct_cm_sockaddr {
+public:
+    test_uct_cm_sockaddr_stress() : m_clients_num(0),
+                                    m_ep_init_disconnect_cnt(0) {
+    }
+
+    void init() {
+        test_uct_cm_sockaddr::init();
+
+        m_clients_num = ucs_max(2, 100 / ucs::test_time_multiplier());
+    }
+
+    int get_ep_index (uct_ep_h ep) {
+        for (int i = 0; i < (2 * m_clients_num); i++) {
+            if (m_all_eps[i].ep == ep) {
+                return i;
+            }
+        }
+
+        return -1;
+    }
+
+    void common_test_disconnect(uct_ep_h ep) {
+        int index;
+
+        index = get_ep_index(ep);
+        EXPECT_GE(index, 0);
+        EXPECT_LT(index, (2 * m_clients_num));
+
+        pthread_mutex_lock(&m_lock);
+        m_all_eps[index].state |= TEST_EP_FLAG_DISCONNECT_CB_INVOKED;
+        pthread_mutex_unlock(&m_lock);
+
+        if (m_all_eps[index].state & TEST_EP_FLAG_DISCONNECT_INITIATOR) {
+            pthread_mutex_lock(&m_lock);
+            m_ep_init_disconnect_cnt--;
+            pthread_mutex_unlock(&m_lock);
+        } else {
+            ASSERT_UCS_OK(uct_ep_disconnect(ep, 0));
+        }
+    }
+
+    static void server_disconnect_cb(uct_ep_h ep, void *arg) {
+        test_uct_cm_sockaddr_stress *self =
+                        reinterpret_cast<test_uct_cm_sockaddr_stress *>(arg);
+
+        self->common_test_disconnect(ep);
+        self->m_server_disconnect_cnt++;
```

Comment:
how come m_server_disconnect_cnt is accessed without lock here?

Reply:
why does it need a lock? it is incremented only here (and never decremented).
same for client below.

Reply:
is it always called from same thread?

Reply:
yes, the disconnect cb is called from the async thread both for rdmacm and tcp_sockcm since both of them process the events in the async thread.

Reply:
i see
still, for consistency, i'd add lock here